src/Reflection/FunctionReflection.php

```diff
@@ -135,7 +135,7 @@ public function getPrototype($format = FunctionReflection::PROTOTYPE_AS_ARRAY)
         $parameters = $this->getParameters();
         foreach ($parameters as $parameter) {
             $prototype['arguments'][$parameter->getName()] = [
-                'type'     => $parameter->getType(),
+                'type'     => $parameter->detectType(),
```

Comment:
@weierophinney doesn't need to be moved out of the component, just out of this class, IMO (future change)